packages/nimble-components/src/select/index.ts

```diff
@@ -94,6 +98,9 @@ export class Select
     @attr({ attribute: 'filter-mode' })
     public filterMode: FilterMode = FilterMode.none;
 
+    @attr({ attribute: 'clearable', mode: 'boolean' })
```

Comment:
I'm not keen on providing a semantic that suggests that it is _always_ visible or not since that is not the behavior. This particular term was lifted from another design system's API for the same functionality (though I understand that that is not a convincing argument on its own...merely a datapoint). I'll leave this thread open for discussion.

Reply:
We don't have too many precedents for `*-hidden` and `*-visible`, so there may be some decision room on this one. I don't think it would be misleading or incorrect to use `clear-button-visible` in this case. Our anchor has an `underline-hidden` attribute, but still shows an underline on hover regardless. If this attribute also provided a way to clear the value other than pressing the button, or if we thought we might change the UI for clearing, I would be sold on `clearable`.

Reply:
If we allow ESC to clear selection as described in the IxD spec then this attribute should also control whether that's allowed. Then I agree that we wouldn't want to use the `*-hidden`/`*-visible` pattern.

Just to make sure we thought about it:
1. we want the default to be not clearable? I think that's reasonable because it would be a behavior change otherwise and because my intuition says that not allowing clear will be fairly common and possibly more common than allowing it.
2. Do we like `clearable` better than `can-clear` or `allow-clear` or other variations? I looked for precedent on native elements and found all of [`cancelable`](https://developer.mozilla.org/en-US/docs/Web/API/Event/cancelable), [`canIntercept`](https://developer.mozilla.org/en-US/docs/Web/API/NavigateEvent/canIntercept), and [`allowFullScreen`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLIFrameElement/allowFullscreen) so I don't know that there's a convention we can follow here. If it's down to personal preference then I'm fine with any of these but would currently vote to stick with `clearable`.

Reply:
1. Yep, default is not `clearable` for the reasons you list.
2. I'm indifferent to your options, and would also just vote to stick to `clearable`.

Reply:
Didn't see any concerning [things named clearable on mdn](https://developer.mozilla.org/en-US/search?q=clearable) and there is discussion of a [`clearable` or `resettable` attribute in old whatwg mailing lists](https://github.com/search?q=org%3Awhatwg%20clearable&type=code) but they are decades old. Closest non-standard thing I see is for the search element and input[type="search"] there is the [`-webkit-search-cancel-button`](https://developer.mozilla.org/en-US/docs/Web/CSS/::-webkit-search-cancel-button) pseudoelement, i.e. "cancel".